target_chains/solana/sdk/js/pyth_solana_receiver/src/index.ts

```diff
@@ -0,0 +1,287 @@
+import { AnchorProvider, Program } from "@coral-xyz/anchor";
+import { Connection, Signer, VersionedTransaction } from "@solana/web3.js";
+import { PythSolanaReceiver } from "./idl/pyth_solana_receiver";
+import Idl from "./idl/pyth_solana_receiver.json";
+import {
+  WormholeCoreBridgeSolana,
+  IDL as WormholeCoreBridgeSolanaIdl,
+} from "./idl/wormhole_core_bridge_solana";
+import {
+  DEFAULT_RECEIVER_PROGRAM_ID,
+  DEFAULT_WORMHOLE_PROGRAM_ID,
+  getConfigPda,
+  getGuardianSetPda,
+  getTreasuryPda,
+} from "./address";
+import { PublicKey, Keypair } from "@solana/web3.js";
+import {
+  parseAccumulatorUpdateData,
+  parsePriceFeedMessage,
+} from "@pythnetwork/price-service-sdk";
+import {
+  DEFAULT_REDUCED_GUARDIAN_SET_SIZE,
+  DEFAULT_TREASURY_ID,
+  POST_UPDATE_ATOMIC_COMPUTE_BUDGET,
+  POST_UPDATE_COMPUTE_BUDGET,
+  VAA_SPLIT_INDEX,
+  VAA_START,
+  VERIFY_ENCODED_VAA_COMPUTE_BUDGET,
+} from "./constants";
+import { Wallet } from "@coral-xyz/anchor/dist/cjs/provider";
+import { getGuardianSetIndex, trimSignatures } from "./vaa";
+import {
+  TransactionBuilder,
+  InstructionWithEphemeralSigners,
+} from "@pythnetwork/solana-utils";
+
+export class PythSolanaReceiverConnection {
+  readonly connection: Connection;
+  readonly wallet: Wallet;
+  readonly provider: AnchorProvider;
+  readonly receiver: Program<PythSolanaReceiver>;
+  readonly wormhole: Program<WormholeCoreBridgeSolana>;
+
+  constructor({
+    connection,
+    wallet,
+  }: {
+    connection: Connection;
+    wallet: Wallet;
+  }) {
+    this.connection = connection;
+    this.wallet = wallet;
+    this.provider = new AnchorProvider(this.connection, this.wallet, {
+      commitment: connection.commitment,
+    });
+    this.receiver = new Program<PythSolanaReceiver>(
+      Idl as PythSolanaReceiver,
+      DEFAULT_RECEIVER_PROGRAM_ID,
+      this.provider
+    );
+    this.wormhole = new Program<WormholeCoreBridgeSolana>(
+      WormholeCoreBridgeSolanaIdl as WormholeCoreBridgeSolana,
+      DEFAULT_WORMHOLE_PROGRAM_ID,
+      this.provider
+    );
+  }
+
+  async withPriceUpdate(
```

Comment:
I think a user gets lost to have a price service connection and this connection. Maybe this class contains a connection and abstracts away all the details from the user?

A couple of other points:
1. passing the update data as Buffer is better as it's a binary blob (whereas string requires knowledge about encoding).
2. I would carefully decide which function is public and which is private to ensure people don't get confused and use the right method.

Reply:
We discussed this and we're gonna keep it separate for now